number_search.py

```diff
@@ -3,25 +3,46 @@
 # use regex to match numbers and fractions
 import re
 
+
 def count_numbers(text):
     """
         counts whole number and decimals i.e -ve and +ve whole numbers decimals
         TODO update doc strings
     """
+    count = 0
+    pattern = re.compile(r'(\d+| \d+\.\d+)')
+    with open('article.txt', 'r') as f:
+        file = f.read()
```

Comment:
try to convert the sting "file" to list of strings by splitting it by space using .split() method.. increase your accuracy as regex doesn't know when to stop.. in a long text if it comes across 23b it will match the 23 and leave the b which is not accurate..